### PR TITLE
Implement Ctrl-O for insert mode

### LINKED ISSUE #181: Temporarily enter command mode from insert mode with ctrl+o
As I gather, this plugin „unbinds self from the type event“. Which I like in general… except: What would it take to allow `ctrl+o` to temporarily enter command mode from insert mode? It seems to work in principle with other mappings such as `ctrl+r`.

### PR DESCRIPTION
**Pull request by QAston**
Fixes #181

### CODE REVIEW

**Comment by asvetliakov** [comment]
Thank you for the PR! Can you do a quick test:
1. Start with new document with single line with content `test`
2. place the cursor after `e` and type `123`
3. press `h`
4. the cursor must be placed after `2`

Another test
1. start with 1 & 2 from previous step
2. press cursor left key 2 times, so the cursor would be after `1`
3. press `<C-o>` and `x`

**Comment by QAston** [comment]
First test:
ctrl-o-h
Second test:
ctrl-o-x

Looks like some of the preexisting tests broke, I'll take a look at that later

**Comment by asvetliakov** [comment]
Can you please add these two scenarios to tests? Thanks!

**Comment by QAston** [comment]
@asvetliakov Hi, I've added the tests and they work fine, but for some reason "Cursor is ok for multiple cursor updates - issue #141" test in visual-modes.test.ts is failing. Any idea why that would be? The changes don't seem to be used in that test at all.

**Comment by QAston** [comment]
Well nvm, the build passed, looks like the test was just flaky

File: package.json [hunk 1]
```diff
@@ -723,6 +723,11 @@
                 "key": "ctrl-u",
                 "when": "editorTextFocus && neovim.mode == insert && !neovim.recording && neovim.ctrlKeysInsert"
             },
+            {
+                "command": "vscode-neovim.ctrl-o-insert",
+                "key": "ctrl+o",
+                "when": "editorTextFocus && neovim.mode == insert && neovim.ctrlKeysInsert"
```
Comment (theol0403) [inline package.json:729]:
```suggestion
                "when": "editorTextFocus && neovim.mode == insert && !neovim.recording && neovim.ctrlKeysInsert"
```

File: README.md [hunk 2]
```diff
@@ -412,6 +412,7 @@ Enabled by `useCtrlKeysForInsertMode = true` (default true)
 | -------------------------- | ---------------------------------------------------------------- | --------------------------------- |
 | `CTRL-r [0-9a-z"%#*+:.-=]` | Paste from register                                              | Works                             |
 | `CTRL-a`                   | Paste previous inserted content                                  | Works                             |
+| `CTRL-o`                   | Switch to normal mode for a single command, then back            | Works                             |
```
Comment (theol0403) [inline README.md:415]:
```suggestion
| <kbd>C-o</kbd>                              | Switch to normal mode for a single command, then back.            | Works                             |
```

**Review by theol0403 — APPROVED** [review]
This has been working great for me. All we need is an approval from @asvetliakov and I can merge.

**Comment by theol0403** [comment]
Could you also rebase onto the new master's readme, and then force push? The suggestions I made have the proper formatting, but I guess git still does not know how to merge without conflict.

**Comment by theol0403** [comment]
I'm going to merge this. It works amazingly for me, the tests pass, and even if there is some edge case or bug it is better than nothing. It should not have any side effect either.

Thank you so much for the PR!
